File: SustitucionMOAModel/Entities/Notificacion.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SustitucionMOAModel.Entities
{
    public class Notificacion
    {
        [Key]
        public int Id { get; set; }

        public string Nombre { get; set; }

        public DateTime FechaInicio { get; set; }

        public DateTime? FechaFin { get; set; }

        public bool Habilitada { get; set; }

        public bool Borrada { get; set; }

        public string LinkAdjunto { get; set; }

        public string Mensaje { get; set; }

        [InverseProperty("NotificacionesAsociadas")]

        public virtual ICollection<Rol> FiltroRoles { get; set; } 


    }
}

File: SustitucionMOA/src/app/common/models/notificacion.ts
import { Adjuntos } from "./adjuntos";
import { Rol } from "./rol";
import { TipoUsuario } from "./tipoUsuario";


export class Notificacion {
    public Id: number;
    public Nombre: string;
    public FechaInicio: Date;
    public FechaFin: Date;
    public HoraInicio: number;
    public Habilitada: boolean;
    public Borrada: boolean;
    public LinkAdjunto: string;
    public Mensaje: string;
    public FiltroRoles: Array<Rol>
    public Prioridad: number;
    public FechaCreacion: Date;
    public ArchivosAdjuntos: Array<Adjuntos>
    constructor() { 
        this.FiltroRoles = new Array<Rol>();
    }
}
